nzshm_model/model.py

```diff
@@ -33,92 +28,69 @@ def __init__(
+        gmm_json: Union[str, Path],
+        hazard_config: 'HazardConfig',
+    ) -> 'NshmModel':
         """
-        the source logic tree for this model.
+        Create a new NshmModel instance from files.
 
-        Returns:
-            a source_logic_tree
-
-        """
-        data = self._slt_data
-        ltv = data.get("logic_tree_version")
-        if ltv is None:  # original json is unversioned
-            return SourceLogicTree.from_source_logic_tree(SourceLogicTreeV1.from_dict(data))
-        elif ltv == 2:
-            return SourceLogicTree.from_dict(data)
-        raise ValueError("Unsupported logic_tree_version.")
-
-    @property
-    def gmm_logic_tree_from_xml(self) -> GMCMLogicTree:
+        NB library users will typically never use this, rather they will obtain a model instance
+        using static method: `get_model_version`.
         """
-        the ground motion logic tree for this model.
-
-        Returns:
-            a gmcm_logic_tree
 
-        """
-        warnings.warn("use NshmModel.gmm_logic_tree instead", DeprecationWarning)
-        return self.psha_adapter(provider=OpenquakeGMCMPshaAdapter).gmcm_logic_tree_from_xml(  # type: ignore
-            self._gmm_xml
-        )
+        # backwards compatatilbity for v1 SourceLogicTree
+        # v1 is not versioned
+        data = NshmModel._slt_data_from_file(slt_json)
+        if data.get("logic_tree_version") is None:
+            source_logic_tree = NshmModel._source_logic_tree_from_v1_json(slt_json)
 
-    @property
-    def gmm_logic_tree(self) -> GMCMLogicTree:
-        """
-        the ground motion logic tree for this model.
+        source_logic_tree = SourceLogicTree.from_json(slt_json)
+        gmcm_logic_tree = GMCMLogicTree.from_json(gmm_json)
+        return cls(version, title, source_logic_tree, gmcm_logic_tree, hazard_config)
 
-        Returns:
-            a gmcm_logic_tree
+    @staticmethod
+    def _slt_data_from_file(filepath: Union[str, Path]) -> Dict[Any, Any]:
+        with Path(filepath).open('r') as jsonfile:
+            data = json.load(jsonfile)
+        return data
 
+    @staticmethod
+    def _source_logic_tree_from_v1_json(filepath: Union[str, Path]) -> SourceLogicTree:
         """
-        data = self._glt_data
-        return GMCMLogicTree.from_dict(data)
+        Create a SourceLogicTree from an old v1 json file; convert to new type.
 
-    def gmm_logic_tree_nrml(self) -> "OQLogicTree":
-        """
-        the ground motion characterisation model (gmcm) logic tree for this model as a OpenQuake nrml compatiable type.
+        Arguments:
+            filepath: the path to the json file specifying the source logic tree
 
         Returns:
-            a gmm_logic_tree.
+            a source logic tree
+
         """
-        doc = NrmlDocument.from_xml_file(self._gmm_xml)
-        return doc.logic_trees[0]
+        return SourceLogicTree.from_source_logic_tree(SourceLogicTreeV1.from_json(filepath))
 
     @classmethod
```

Comment:
yetanother @staticmethod

Reply:
agreed to continue to use `@classmethod` to preserve pattern and make method's application more obvious